helix-vcs/src/git.rs

```diff
@@ -88,6 +241,90 @@ impl DiffProvider for Git {
         }
         Some(data)
     }
+
+    fn get_changed_files(&self, cwd: &Path) -> Result<Vec<FileChange>> {
+        Self::status(
+            &Self::open_repo(cwd, None)
+                .ok_or_else(|| anyhow::anyhow!("no Git repository found"))?
+                .to_thread_local(),
+        )
+    }
+}
+
+impl RawChanges {
+    pub fn add_addition(&mut self, addition: RawAddition) {
+        self.additions.push(addition);
+    }
+
+    pub fn add_deletion(&mut self, deletion: RawDeletion) {
+        match self.deletions.entry(deletion.oid) {
+            Entry::Occupied(entry) => {
+                entry.into_mut().push(deletion);
+            }
+            Entry::Vacant(entry) => {
+                entry.insert(vec![deletion]);
+            }
+        }
+    }
+
+    pub fn add_modification(&mut self, modification: RawModification) {
+        self.modifications.push(modification);
+    }
+}
+
+impl From<RawChanges> for Vec<FileChange> {
+    // Unlike Git, we only look for pure renames at the moment.
+    // TODO: detect renames with minor changes
+    fn from(mut raw: RawChanges) -> Self {
+        let mut status_entries = vec![];
+
+        let additions_left = if !raw.additions.is_empty() && !raw.deletions.is_empty() {
```

Comment:
Inverting the condition would mean having the one-line `else` branch at the top and the big branch below, making it much easier to remember the condition when getting to the `else`